src/bitcoin_wallet.rs

```diff
@@ -62,6 +65,20 @@ impl Wallet {
         Ok(())
     }
 
+    pub fn random_transient_sk(&self) -> anyhow::Result<SecretKey> {
```

Comment:
I guess this is for test only? If so add `#[cfg(test)]`.

If not... Shouldn't we derive it from our seed?

Reply:
It's currently not used for testing...

> If not... Shouldn't we derive it from our seed?

In a way it is derived from the seed, since the Bitcoin wallet's private key is derived from the seed :stuck_out_tongue: Then we need to hash it with something unique so that we don't use the same one for all swaps. In `cnd` we use a swap ID, but when I call it in Nectar we still did not have one. That's why I hashed it with a random string. This function is only called once per swap anyway (then saved to a `HashMap` inside `network::Nectar`).

I think it's bad though. That's why I left the TODO just below.

Reply:
Yeah, thinking about it, I think we should just use bip32 to derive those keys. Maybe under a different account or path than what bitcoind does.

Reply:
That sounds very good! I would say we should do that in a separate PR though.